Source/Core/VideoBackends/Metal/MTLMain.mm

```diff
@@ -0,0 +1,144 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "VideoBackends/Metal/VideoBackend.h"
+
+#include <AppKit/AppKit.h>
+#include <Metal/Metal.h>
+#include <QuartzCore/QuartzCore.h>
+
+#include "Common/Common.h"
+#include "Common/MsgHandler.h"
+
+#include "VideoBackends/Metal/MTLObjectCache.h"
+#include "VideoBackends/Metal/MTLPerfQuery.h"
+#include "VideoBackends/Metal/MTLRenderer.h"
+#include "VideoBackends/Metal/MTLStateTracker.h"
+#include "VideoBackends/Metal/MTLUtil.h"
+#include "VideoBackends/Metal/MTLVertexManager.h"
+
+#include "VideoCommon/FramebufferManager.h"
+#include "VideoCommon/VideoCommon.h"
+#include "VideoCommon/VideoConfig.h"
+
+std::string Metal::VideoBackend::GetName() const
+{
+  return NAME;
+}
+
+std::string Metal::VideoBackend::GetDisplayName() const
+{
+  // i18n: Apple's Metal graphics API (https://developer.apple.com/metal/)
+  return _trans("Metal");
+}
+
+bool Metal::VideoBackend::Initialize(const WindowSystemInfo& wsi)
+{
+  @autoreleasepool
+  {
+    if (wsi.type != WindowSystemType::MacOS || !wsi.render_surface)
```

Comment:
If we keep the iOS support in, should probably `ifdef` this check out on iOS. In my fork, I have a `WindowSystemType::iOS`.

Reply:
I feel like it seems reasonable to just add `WindowSystemType::iOS` here once it gets added to the enum

Maybe add a dedicated `static bool WindowSystemTypeSupportsMetal(WindowSystemType type)` function for that?

Reply:
Yeah, I think that's a good idea. Then in my fork I can just create an `ifdef` for this enum type with minimal chances of a merge conflict.